agent/agents/supervisor/supervisor.go

```diff
@@ -160,16 +160,36 @@ func (s *Supervisor) AgentsLogs() map[string][]string {
 
 	for id, agent := range s.agentProcesses {
 		newID := strings.ReplaceAll(id, "/agent_id/", "")
-		res[fmt.Sprintf("%s %s", agent.requestedState.Type.String(), newID)] = agent.logStore.GetLogs()
+		res[fmt.Sprintf("%s %s", agent.requestedState.Type.String(), newID)], _ = agent.logStore.GetLogs()
 	}
 
 	for id, agent := range s.builtinAgents {
 		newID := strings.ReplaceAll(id, "/agent_id/", "")
-		res[fmt.Sprintf("%s %s", agent.requestedState.Type.String(), newID)] = agent.logStore.GetLogs()
+		res[fmt.Sprintf("%s %s", agent.requestedState.Type.String(), newID)], _ = agent.logStore.GetLogs()
 	}
 	return res
 }
 
+// AgentLogByID returns logs by Agent ID.
+func (s *Supervisor) AgentLogByID(id string) ([]string, uint) {
+	s.rw.RLock()
+	defer s.rw.RUnlock()
+	s.arw.RLock()
+	defer s.arw.RUnlock()
```

Comment:
I think we don't need this lock here since we don't read/write from the `lastStatuses` list. The same for the method above.

Reply:
@BupycHuk you right, done